nrich-search/README.md

```diff
@@ -84,55 +106,62 @@ public class Car {
     @ManyToOne(fetch = FetchType.LAZY)
     private CarType carType;
 }
-
-
 ```
 
-looks like this (if both standard and quick search functionality is required)
+we have to ensure that its JPA repository extends `SearchExecutor` interface like this:
 
 ```java
-
-
-public interface CarRepository extends JpaRepository<Car, Long>, SearchExecutor<Car>, StringSearchExecutor<Car> {
+public interface CarRepository extends JpaRepository<Car, Long>, SearchExecutor<Car> {
 
 }
+```
 
+Then, if we need to query car entities by its:
 
-```
+- registration number
+- manufacture date (time period from-to)
+- price (range query from-to, from included)
+- number of kilometers
+- car type make
+- car type model
 
-When searching for Car instances using `SearchExecutor<Car>` it is necessary to define a class holding all properties that will be searched. In this case that would look something like this
-(`BaseSortablePageableRequest` is a class provided by the module, but it is not strictly necessary to extend it just makes it easier to handle paging and sorting parameters):
+we can define a class that describes wanted query parameters:
 
 ```java
-
 @Setter
 @Getter
-public class SearchCarRequest extends BaseSortablePageableRequest {
+public class SearchCarDemoRequest extends BaseSortablePageableRequest {
 
     private String registrationNumber;
 
-    private Date manufacturedDateFrom;
+    private Instant manufacturedTimeFrom;
 
-    private Date manufacturedDateTo;
+    private Instant manufacturedTimeTo;
 
-    private BigDecimal priceFrom;
+    @DecimalMin("0.00")
+    private BigDecimal priceFromIncluding;
 
+    @DecimalMin("0.00")
     private BigDecimal priceTo;
 
+    @Min(0)
     private Integer numberOfKilometers;
 
+    @Size(max = 20)
     private String carTypeMake;
 
+    @Size(max = 40)
     private String carTypeModel;
 
 }
-
 ```
 
-Assuming we want to return a projection instead of Car entities then we would also define a projection class:
+`BaseSortablePageableRequest` is a class provided by the module that makes paging and sorting easier by including page and sort properties.
+It is not necessary to extend it.
```

Comment:
Maybe expand to `It is not necessary to extend it for search to work` for clarity.

Reply:
Also done.